hack/version.py

```diff
@@ -31,142 +25,135 @@
+    commit = hive_repo.rev_parse(branch_arg)
+    master_commit = find_branch(hive_repo, HIVE_BRANCH_DEFAULT)
+
+    # This includes when the commit is equal to the master commit
+    if not hive_repo.is_ancestor(commit, master_commit):
+        log(
+            "Commit {} is not equal or and ancestor of the master branch".format(
+            commit.hexsha)
+        )
+        sys.exit(1)
+    return commit.hexsha, HIVE_VERSION_PREFIX
 
-    # Was (a commit on) an ocm-X.Y[-mce-M.N] branch specified?
-    # This will find all such branch names and add them to the channel list.
-    # However, we'll only support >1 entry in this list if we're tracking
-    # master, because otherwise which would we use to compute the semver?
-    channels = []
-    maybe_prefix = None
+
+
+def process_current_branch(hive_repo, branch_arg):
+    """Validate and process the input (or default) branch.
+
+    :param hive_repo: The git.Repo for the local hive clone.
+    :param branch_arg: The string commit-ish input representing the tip of a valid master or mce-* branch.
+    :return commit_hash: The canonical full-length sha of the commit corresponding to branch_arg.
+    :return prefix: The two-digit semver prefix of the bundle version to use. If branch_arg is
+        (a descendant of) master, we'll use HIVE_VERSION_PREFIX. If not, and it names an 'mce-X.Y'
+        branch, we'll use X.Y.
+    :raise: If we get a branch_arg that's invalid (doesn't correspond to a real commit in hive_repo),
+        or that's not 'mce-X.Y' or master.
+    """
+    # This will raise an exception if there's no such commit-ish
+    commit = hive_repo.rev_parse(branch_arg)
+
+    prefix = None
+    maybe_prefix = []
     for info in remote_branch_info(hive_repo).values():
-        if info["commit_hash"] == commit_hash:
-            channels.append(info["channel"])
-            # We'll only use this if we end up with one
-            # entry in this list AND we're not tracking master
-            maybe_prefix = info["prefix"]
-
-    if commit_hash == master_hash or (is_master_ancestor and not maybe_prefix):
-        prefix = HIVE_VERSION_PREFIX
-        if commit_hash == master_hash or _mode == "standalone":
-            channels.append(CHANNEL_DEFAULT)
-    else:
-        if len(channels) > 1:
-            raise ValueError(
-                "Found more than one channel candidate at {}: {}.\n".format(
-                    branch_arg, channels
-                )
-                + "I can't handle this unless they're direct descendants of master -- "
-                + "which one would I use as the semver base?"
-            )
-        prefix = maybe_prefix
+        if info["commit_hash"] == commit.hexsha:
+            maybe_prefix.append(info["prefix"])
 
-    if not channels or not prefix:
-        if _mode != "standalone":
+    if len(maybe_prefix) > 1:
+        try:
+            branch_arg = repo.active_branch.name
+        except TypeError:
```

Comment:
What causes this TypeError? If the active_branch is unnamed?

Reply:
Bear with me: Let's say you run the script and your HEAD is on another branch that you created locally, but it's two commits behind the tip of this new local branch. That commit (HEAD) also corresponds to the tip of the master branch and the tip of mce-2.2. Now, this commit does not have a corresponding branch name and it is ambigious as to which branch name version to use. This is unlikely to happen but good to handle nonetheless.

Reply:
How does that result in a TypeError though?

Reply:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/lleshchi/.local/lib/python3.11/site-packages/git/repo/base.py", line 896, in active_branch
    return self.head.reference
           ^^^^^^^^^^^^^^^^^^^
  File "/home/lleshchi/.local/lib/python3.11/site-packages/git/refs/symbolic.py", line 309, in _get_reference
    raise TypeError("%s is a detached symbolic reference as it points to %r" % (self, sha))
TypeError: HEAD is a detached symbolic reference as it points to 'cef3cb0773f460247c60c666a33c6ec090522e8e'
```

Reply:
Okay, detached HEAD, got it :+1: